File: app/src/main/java/com/paytm/hpclpos/fragmentnoncardedtransaction/ccmsrecharge/cheque/ChequeMainFragment.java
package com.paytm.hpclpos.fragmentnoncardedtransaction.ccmsrecharge.cheque;

import android.os.Bundle;
import android.text.InputType;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.EditorInfo;
import android.view.inputmethod.InputConnection;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.Toast;

import androidx.fragment.app.Fragment;

import com.paytm.hpclpos.R;
import com.paytm.hpclpos.constants.EnterMobileNoKeyboard;
import com.paytm.hpclpos.constants.Validation;

public class ChequeMainFragment extends Fragment {

 private EditText chequeNoEditText;
 private EditText micrCodeEditText;

    public ChequeMainFragment() {
        // Required empty public constructor
    }


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.d("onCreate","ChequeMainFragment");

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment

        final View rootView = inflater.inflate(R.layout.fragment_cheque_main, container, false);

        rootView.setOnTouchListener((view, motionEvent) -> true);

        chequeNoEditText = rootView.findViewById(R.id.etChqNo);
        micrCodeEditText = rootView.findViewById(R.id.etMicrCode);
        LinearLayout gotoBack= rootView.findViewById( R.id.gotoBack);
        EnterMobileNoKeyboard mobNokeyboard = rootView.findViewById(R.id.keyboard);

        LinearLayout layoutDone = mobNokeyboard.findViewById(R.id.llayout_done);


        layoutDone.setOnClickListener(v -> callRechargeAmountFrag());

        gotoBack.setOnClickListener(v -> getActivity().onBackPressed());

        chequeNoEditText.setOnTouchListener((v, event) -> {
            int inType = chequeNoEditText.getInputType(); // backup the input type
            chequeNoEditText.setInputType(InputType.TYPE_NULL); // disable soft input
            chequeNoEditText.onTouchEvent(event); // call native handler
            chequeNoEditText.setInputType(inType); // restore input type
            // prevent system keyboard from appearing when EditText is tapped
            chequeNoEditText.setRawInputType(InputType.TYPE_CLASS_TEXT);
            chequeNoEditText.setTextIsSelectable(true);

            // pass the InputConnection from the EditText to the keyboard
            InputConnection ic = chequeNoEditText.onCreateInputConnection(new EditorInfo());
            mobNokeyboard.setInputConnection(ic);
            return true; // consume touch even
        });




        micrCodeEditText.setOnTouchListener((v, event) -> {
            int inType = micrCodeEditText.getInputType(); // backup the input type
            micrCodeEditText.setInputType(InputType.TYPE_NULL); // disable soft input
            micrCodeEditText.onTouchEvent(event); // call native handler
            micrCodeEditText.setInputType(inType); // restore input type

            // prevent system keyboard from appearing when EditText is tapped
            micrCodeEditText.setRawInputType(InputType.TYPE_CLASS_TEXT);
            micrCodeEditText.setTextIsSelectable(true);

            // pass the InputConnection from the EditText to the keyboard
            InputConnection icNew = micrCodeEditText.onCreateInputConnection(new EditorInfo());
            mobNokeyboard.setInputConnection(icNew);

            return true; // consume touch even
        });

        return rootView;
    }


    private void callRechargeAmountFrag(){
        String cheqNo = chequeNoEditText.getText().toString();
        String mICRCode = micrCodeEditText.getText().toString();

        if (!Validation.isvalidChequeNo(cheqNo.trim())) {

            Toast.makeText(getContext(),"Please enter a valid Cheque No",Toast.LENGTH_SHORT).show();


        } else if (!Validation.isvalidMICRCode(mICRCode.trim())) {
            Toast.makeText(getContext(),"Please enter a valid MICR Code",Toast.LENGTH_SHORT).show();
        }

        else {

            Fragment fragment = new CheqEnterCntrlCardNoFrag();
            getActivity().getSupportFragmentManager().beginTransaction().setCustomAnimations(R.anim.slide_from_right, R.anim.slide_to_left)
                    .replace(android.R.id.content, fragment, fragment.getClass().getSimpleName()).addToBackStack(null).commit();
            chequeNoEditText.setText("");
            micrCodeEditText.setText("");
        }

    }
}

File: app/src/main/java/com/paytm/hpclpos/ui/checkandmicrnumber/ChequeMainFragment.kt
package com.paytm.hpclpos.ui.checkandmicrnumber

import android.content.Context
import android.os.Bundle
import android.text.InputType
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.inputmethod.EditorInfo
import android.view.inputmethod.InputMethodManager
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import com.paytm.hpclpos.R
import com.paytm.hpclpos.constants.EnterMobileNoKeyboard
import com.paytm.hpclpos.constants.GlobalMethods
import com.paytm.hpclpos.constants.Validation
import com.paytm.hpclpos.databinding.FragmentChequeMainBinding
import com.paytm.hpclpos.posterminal.base.BaseFragment

class ChequeMainFragment : BaseFragment() {

    var mobNokeyboard: EnterMobileNoKeyboard? = null
    lateinit var binding: FragmentChequeMainBinding
    override fun updateTimerUi(l: Long) {
        requireView().findViewById<TextView>(R.id.timer).setText(l.toString()
            .substring(0, l.toString().length - 3) +" s")
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_cheque_main, container, false)
        binding.root.setOnTouchListener { view, motionEvent -> true }
        viewFinds()
        return binding.root
    }

    private fun viewFinds() {
        handleEnterKey()
        binding.gotoBack.setOnClickListener {
            hideKeypad()
            requireActivity().onBackPressed()
        }
    }

    private fun callControlCardNoFrag() {
        if(binding.etChqNo.text.toString().isBlank() && binding.etMicrCode.text.toString().isBlank()) {
            binding.etChqNo.setError( resources.getString(R.string.input_required))
            binding.etMicrCode.setError( resources.getString(R.string.input_required))
        } else if(binding.etChqNo.text.toString().isBlank()) {
            binding.etChqNo.setError( resources.getString(R.string.input_required))
        } else if(binding.etMicrCode.text.toString().isBlank()) {
            binding.etMicrCode.setError( resources.getString(R.string.input_required))
        } else if(!Validation.isvalidChequeNo(binding.etChqNo.text.toString())) {
            Toast.makeText(context, resources.getString(R.string.entervalidcheqno), Toast.LENGTH_LONG).show()
            binding.etChqNo.setError( resources.getString(R.string.entervalidcheqno))
        } else if(!Validation.isvalidMICRCode(binding.etMicrCode.text.toString())) {
            Toast.makeText(context, resources.getString(R.string.entervalidmicrno), Toast.LENGTH_LONG).show()
            binding.etMicrCode.setError( resources.getString(R.string.entervalidmicrno))
        } else {
            GlobalMethods.setChequeNumber(binding.etChqNo.text.toString())
            GlobalMethods.setMicrCode(binding.etMicrCode.text.toString())
            if (GlobalMethods.getCardInfoEntity() != null && GlobalMethods.getCardInfoEntity()!!.isControlCardNumber)
                navController!!.navigate(R.id.controlCardNumber)
            else {
                navController!!.navigate(R.id.action_amountEntryFragment)
            }
        }
    }

    private fun handleEnterKey() {
        val imgr: InputMethodManager =
            requireActivity().getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        binding.etChqNo.requestFocus()
        imgr.toggleSoftInput(InputMethodManager.SHOW_FORCED, InputMethodManager.HIDE_IMPLICIT_ONLY)
        binding.etMicrCode.setImeOptions(EditorInfo.IME_ACTION_DONE);
        binding.etMicrCode.setOnEditorActionListener({ v, actionId, event ->
            if (actionId == EditorInfo.IME_ACTION_DONE) {
                closesoftKeypad(binding.etMicrCode)
                callControlCardNoFrag()
                true
            } else false
        })
    }
}